node/node.go

```diff
@@ -389,11 +389,7 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 
 	// only the syncer gets the storage which is online connected
 	chainSyncer := chain.NewDefaultSyncer(&cstOnline, &cstOffline, nodeConsensus, chainStore)
-	chainReader, ok := chainStore.(chain.ReadStore)
```

Comment:
This cast is redundant. The compiler will complain if `chainStore` isn't a `chain.ReadStore`.